src/dune_rules/lib_id.ml

```diff
@@ -34,8 +39,15 @@ module Local = struct
   include T
   include Comparable.Make (T)
 
-  let make ~loc ~src_dir ~enabled_if name = { name; loc; enabled_if; src_dir }
-  let loc t = t.loc
+  let make ~loc ~src_dir ~enabled_if ~visibility name =
+    { name; loc; enabled_if; src_dir; visibility }
+  ;;
+
+  let loc t =
```

Comment:
yeah, I'll do that instead. Sorry for the churn here, I think I ported a version using `best_name` that didn't end up being used, and adding the `loc` is definitely a great idea.

Reply:
actually, we can't do that change in isolation, because then library redirects and libraries with public names will get the same `loc` (`Lib_id.compare` returns `Eq` when it shouldn't). Will either need to track this "best loc" in a separate field or otherwise add more fields to lib_id for the comparison function. I think I prefer adding a `best_loc` field.